### PR TITLE
Handle expanding row content resizes

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- if using sandbox: change this line to "demonstration": https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js#L51
- add an app
- open the deployment flow
- click on the credential and then "Add credential..."
- click on the authentication type
- click "oauth2"
- the form should appear and the container should animate to accommodate the extra length
- change it back to jsonfile and the container should animate again

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5918/
Test PASSed.

File: jujugui/static/gui/src/app/components/expanding-row/test-expanding-row.js [hunk 1]
```diff
@@ -160,4 +160,23 @@ describe('ExpandingRow', () => {
...
+
+  it('can stop observing the DOM when unmounted', () => {
+    var renderer = jsTestUtils.shallowRender(
+      <juju.components.ExpandingRow
+        clickable={false}>
+        <span>closed</span>
+        <span>open</span>
+      </juju.components.ExpandingRow>, true);
+    var instance = renderer.getMountedInstance();
+    // Mock the ref. The MutationObserver needs a real DOM node.
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/expanding-row/test-expanding-row.js]: Really nice test, thanks for this comments.

**Comment by hatched** [comment]
:shipit:

**Pull request merged**